migrations/20201122043349_more-mod-data.sql

```diff
@@ -26,29 +26,38 @@ CREATE TABLE side_types (
 INSERT INTO side_types (name) VALUES ('required');
 INSERT INTO side_types (name) VALUES ('no-functionality');
 INSERT INTO side_types (name) VALUES ('unsupported');
+INSERT INTO side_types (name) VALUES ('unknown');
 
 CREATE TABLE licenses (
     id serial PRIMARY KEY,
     short varchar(60) UNIQUE,
     name varchar(1000) UNIQUE
 );
 
-INSERT INTO licenses (short, name) VALUES ('arr', 'All Rights Reserved');
+INSERT INTO licenses (short, name) VALUES ('custom', 'Custom License');
 
 ALTER TABLE versions
     ADD COLUMN featured BOOLEAN NOT NULL default FALSE;
 ALTER TABLE files
-    ADD COLUMN "primary" BOOLEAN NOT NULL default FALSE;
+    ADD COLUMN is_primary BOOLEAN NOT NULL default FALSE;
 
 ALTER TABLE mods
     ADD COLUMN license integer REFERENCES licenses NOT NULL default 1;
 ALTER TABLE mods
     ADD COLUMN license_url varchar(1000) NULL;
 ALTER TABLE mods
-    ADD COLUMN client_side integer REFERENCES side_types NOT NULL default 1;
+    ADD COLUMN client_side integer REFERENCES side_types NOT NULL default 4;
 ALTER TABLE mods
-    ADD COLUMN server_side integer REFERENCES side_types NOT NULL default 1;
+    ADD COLUMN server_side integer REFERENCES side_types NOT NULL default 4;
 ALTER TABLE mods
     ADD COLUMN discord_url varchar(255) NULL;
 ALTER TABLE mods
-    ADD COLUMN slug varchar(255) NULL UNIQUE;
\ No newline at end of file
+    ADD COLUMN slug varchar(255) NULL UNIQUE;
+
+CREATE TABLE downloads (
+    id serial PRIMARY KEY,
+    version_id bigint REFERENCES versions ON UPDATE CASCADE NOT NULL,
+    date timestamptz DEFAULT CURRENT_TIMESTAMP NOT NULL,
+    -- A SHA1 hash of the downloader IP address
+    identifier varchar(40) NOT NULL
```

Comment:
This is not anonymized, and should probably be deleted after a few days.  Most users will be using IPv4 (Do we even support IPv6? We should), so that gives a bit less than 2^32 addresses - this ends up taking <30 minutes to generate the hash of *every* IPv4 address, allowing this to be entirely de-anonymized.

I'm assuming that this is intended to avoid tracking multiple downloads from the same IP in a short time range?  If so, I  would recommend clearing this field for every record older than an hour.  It would also be good to generate a random salt on startup that is hashed with the IP address, and ensure that that salt is never logged or saved, but that doesn't entirely fix the issue; we need to clear old records on a regular basis.

Reply:
Commenting on the IPv6, its becoming slowly standardised, i suggest using it instead of having to rewrite code in the future. You could use IPv4 as a fallback if IPv6 is unavailable or something 🤷🏽

Reply:
`realip_remote_addr` supports IPV6 iirc

Reply:
Ah, nevermind then 😅

Reply:
The IPv6 comment is the fact that our current server hosting stuff doesn't support IPv6.  This is either an issue with the DNS records or the actual host configuration (@Geometrically?)

Reply:
Wdym? The server isn't on IPV6, but the host should accept IPV6 requests